### DISCUSSION TITLE
Can't disable upload to GitHub

### DISCUSSION THREAD

**Discussion by viragtripathi** [jreleaser/jreleaser#343] [Q&A]
Hi, I am trying to disable the final artifact upload but it's not working. I have both publication `<release><github><enabled>false</enabled>..` and upload `<upload><enabled>false</enabled>..` set to `false`. Is there any other configuration I need to enable/disable?

**Comment by aalmiray** [comment]
Well, it depends on what you're trying to accomplish. If you do not want artifacts to be uploaded may I ask what do you want to to with them in the first place?

If you only want a Git release created with a changelog then do not configure any `<distributions>` nor `<files>` sections, that's the easiest way. If you do want distributions because you also make use of packagers then you must publish files somewhere. Your current options are the Git release page, Artifactory, or HTTP.

If you need certain files to be uploaded but others not, then you can define an extra property `skipUpload` as explained at https://jreleaser.org/guide/latest/configuration/upload/artifactory.html#_skip_upload. Note that `upload` != `release`, or rather, upload is independent from release, that is disabling upload does not disable release. If you want to skip certain files from being released then define an extra property named `skipRelease` as explained at https://jreleaser.org/guide/latest/configuration/release/github.html#_skip_release.

These properties may be defined at the `<artifact>` or `<distribution>` elements.

**Reply by viragtripathi** [reply]
Thanks @aalmiray for the detailed response. I agree, I should have provided the full context of my use-case.
I am working with a multi-module project and it's an internal project so I upload the final artifacts to another public distribution repo. I am using maven profiles for the build and jreleaser configurations and it's disabled by default. I don't want to upload the artifacts until all the testing is done and it's ready to be released. I can build and publish a specific module i.e. `mvn -pl :<module> -am -P<profile> package` or enable the profile and do `mvn install` to build and publish all the modules that are configured with jreleaser. 
My requirement is little different as I don't want to make these artifacts available on GitHub until all the testing is performed i.e. I want to test the tarballs locally first and if all is well then do a final build and upload. 

I created another profile without jreleaser and this is activated by default so now I can build locally with `mvn install` or individual modules and use the `jreleaser` profile when I am ready to publish the artifacts to GitHub.

**Reply by aalmiray** [reply]
The jreleaser-maven-plugin does not bind any of its goals to a default lifecycle phase for a reason: to give you total control on when and how they are invoked. You are in charge in deciding when they get invoked. 

If you bind them to the `package` phase then it's very likely that you'll need to use a profile so that the goal is activated exactly when you want it.

**Reply by viragtripathi** [reply]
Thanks @aalmiray. It makes sense and with my requirement, having separate profiles is a better option so that regular building and testing is not affected. Thanks a lot for your responses and clarifications.
